### PR TITLE
Fix and improve auto update

### LINKED ISSUE #4: Auto update may occur during import
When app is auto updated it is restarted, it may happen when an import is ongoing. See if this use case is reproducible and if auto update can be disable during import.

------
**Edit by @fdaveine 21/05:** additionally auto update seem to not be trigger at all on Linux (not yet tested for macOS) 

-------
**Edit by @fdaveine 04/06:** on Windows update work but there is no notification (so the current issue remain), may be related to:
- https://github.com/electron-userland/electron-builder/issues/2700

### PR DESCRIPTION
**Pull request by fdaveine**
Fixes #4 
- Auto-update can now only occurred when app start, before user can use it
- A splash screen with a loader is shown to user during the update process
- Add an advanced menu (hidden by default on Linux and Windows, hit `Alt` to show it) which allow user to find log files easily and to enable DEBUG (log level) mode
- Simplify CI update release process

Useful links:

https://stackoverflow.com/questions/51003995/how-can-i-test-electron-builder-auto-update-flow
https://github.com/electron-userland/electron-builder/issues/3053
https://github.com/electron-userland/electron-builder/issues/3753#issuecomment-474698833
https://github.com/electron-userland/electron-builder/issues/1505
https://github.com/electron-userland/electron-builder/issues/2521#issuecomment-363130128

### CODE REVIEW

File: src/main/appMenuTemplate.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        click() { toggleDebugFileLogLevel(); }
+      },
+      {
+        label: 'Open logs folder',
+        click: () => {
+          let logsFolderPath;
+          if (isMac) {
+            logsFolderPath = path.join(app.getPath("home"), "Library", "Logs", "paper-matter-import-export");
+          } else {
+            logsFolderPath = path.join(app.getPath("appData"), "paper-matter-import-export", "logs");
```
Comment (fdaveine) [inline src/main/appMenuTemplate.js:38]: We should test this path is OK for Windows too (only tested under Linux)

**Comment by fdaveine** [comment]
@antoinelibert to review plz
Additionally could you test the update process on Windows by following the steps in `README.md` > `#### Debug update process` (maybe README.md need to be completed with Windows only steps).

File: src/main/appMenuTemplate.js (hunk 1 above)
Comment (antoinelibert) [reply src/main/appMenuTemplate.js:37]: This is where the logs have been stored on my Windows machine.

image

Comment (fdaveine) [reply src/main/appMenuTemplate.js:37]: This is the same folder opened when you hit `Alt` > menu **Advanced** > **Open logs folder** ?

**Review by antoinelibert — APPROVED** [review]

File: src/main/appMenuTemplate.js (hunk 1 above)
Comment (antoinelibert) [reply src/main/appMenuTemplate.js:37]: Yes.

**Pull request merged**